### ISSUE TITLE
Flow crashes when using `||` or ternary with constructors

### ISSUE THREAD

**Issue by barretron** [facebook/flow#8154]
The following code causes Flow server to crash (simplified example):
```js
let err = new TypeError();
let constructor = typeof err.constructor === 'function' ? err.constructor : Error;
const clone = Object.create(constructor.prototype);
```
Whereas this doesn't:
```js
let err = new TypeError();
let constructor = err.constructor;
if (typeof constructor !== 'function') {
  constructor = Error;
}
const clone = Object.create(constructor.prototype);
```

Flow version: 0.110.1, 0.107.0

## Expected behavior
`No errors!`
## Actual behavior
Flow server crash.
Try Flow console output:
```
add_output: no source for error: EInvalidPrototype (: "empty prototype object"):
```

* Link to Try-Flow or Github repo:

  Crash examples:
  [Try Flow (ternary)](https://flow.org/try/#0DYUwLgBCBO0QvBAdiA7hAKgTwA4gKKwD20AFAJQDcAUKJAMZFIDOY0ArvWCQhGLiCIAzKLAB0jFm07c48eRADkQ9ki4BLJoogB+UdAlNWHLjwBcEQtBI1JrCPWBMQvAPIAjAFYguE6CABDMBBSO2lTAxxrbn48KiA)
  [Try Flow (pipe)](https://flow.org/try/#0DYUwLgBCBO0QvBAdiA7hAKgTwA4gKKwD20AFAJQDcAUKJAMZFIDOY0ArvWCQlLAHSMWbTtzgAfcRELQSNIawj1gTELwDyAIwBWILoOggAhmBCkFIriX45Z3MLhBUgA)

  Working example:
  [Try Flow (if statement)](https://flow.org/try/#0DYUwLgBCBO0QvBAdiA7hAKgTwA4gKKwD20AFAJQDcAUKJAMZFIDOY0ArvWCQlLAHSMWbTt2g0AlgDMIpMLhBEZQ1hy48AhPEQByKeyRcJTHeQgBvahAgqR6uIkLQSNAL7VbN4ExC8A8gBGAFYgXILQIACGYCCktmpi-DjO3PJ4VEA)

**Comment by nmote** [comment]
Thanks for the excellent bug report! @panagosg7 has agreed to take a look.

**Comment by panagosg7** [comment]
Fixed by c66358199e8e578c51215271bfbbbe1514e27cfb
